lib/src/bound_multipart_stream.dart

```diff
@@ -133,14 +133,19 @@ class BoundMultipartStream {
             _subscription.pause();
             _buffer = data;
             _index = 0;
-            _parse();
+            try {
+              _parse();
+            } catch (e, st) {
+              _state = _FAIL;
+              _controller.addError(e, st);
```

Comment:
So, I'd also add:
```dart
_subscription.cancel();
_controller.close();
```
or, if you want to handle any potential error from the _subscription.cancel (that is, unless you can somehow guarantee there won't be one):
```dart
Future.sync(_subscription.cancel)
    .catchError(_controller.addError)
    .whenComplete(_controller.close);
_controller.addError(e, st);
```
(Not sure if there is a better place to surface the error from calling `cancel` than the stream itself)

Reply:
In this case, the controller is configured to cancel on error and `_controller.addError` triggers the `onCancel` callback which then cancels the subscription to the input stream (in `_tryPropagateControllerState`). 

I'll add code to ensure the output controller gets closed when the subscription is cancelled due to error. One thing I am unclear on is the behavior of `addError` when called multiple times before the sink is closed; I will write some code to validate this behavior (unless there is a document that describes it well that I haven't found that you can point me to).

Thank you for the feedback.